### PR TITLE
Allow customizable tailwind config file location

### PR DESCRIPTION
**Pull request by davidtkramer**
First off, thanks for making this library @Arthie! Really great work.

I'm using this package with create-react-app and emotion. I'd like to be able to import my tailwind config and pass it to emotion's theme provider. In order to do this with create-react-app however, the tailwind config has to be under the `src` folder. 

This PR adds the ability to specify a custom config file path using babel-plugin-macros' `config` option. This allows the user to specify a `configFilePath` under the tailwindcssinjs config. For example, in package.json you can do:

```json
"babelMacros": {
 "tailwindcssinjs": {
    "configFilePath": "src/tailwind.config.js"
  }
}
```

Let me know what you think! 🙂 If you are okay with this change I can include updated documentation on this PR as well.

### CODE REVIEW

File: packages/macro/src/babel-plugin-macros.d.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import * as _babelPluginMacros from "babel-plugin-macros";
+
+declare module "babel-plugin-macros" {
+  interface MacroParams {
+    config: any;
+  }
+}
```
Comment (davidtkramer) [inline packages/macro/src/babel-plugin-macros.d.ts:7]: The types for babel-plugin-macros are missing typings for the `config` argument, so I've had to workaround that via module augmentation. We can submit a PR to DefinitelyTyped to get these types updated.

**Comment by Arthie** [comment]
Hello @davidtkramer, great idea! This is definitely something we should add.

I am finishing up on some mayor internal changes in the macro and tailwind-data package.
The changes will be pushed to the master branch soon! Unfortunately, this will require some changes to this PR before I can merge it.

(The changes I am working on will improve the developer experience significantly. I'm working on adding support for hot reloading and instant feedback on changes made in the tailwind config)

**Comment by davidtkramer** [comment]
@Arthie, sounds good! I'm excited for hot reloading, that'll be great. I can revisit this once your changes have been pushed 👍

**Comment by Arthie** [comment]
@davidtkramer Glad, you are excited! When your PR is merged I'll release @tailwindcssinjs/macro v0.5.0!

Can u change the PR to make these changes in `packages/macro/src/macro.ts`
```typescript
interface TailwindcssinjsMacroParams extends MacroParams {
  config?: {
    configPath: string; //<==
    experimentalDevelopmentMode: boolean;
  };
}

function tailwindcssinjsMacro({
  references: { default: paths },
  state,
  babel: { types: t, template },
  config = {
    configPath: "./tailwind.config.js", //<==
    experimentalDevelopmentMode: false,
  },
}: TailwindcssinjsMacroParams) {
  try {
    state.dev =
      process.env.NODE_ENV === "development" &&
      config.experimentalDevelopmentMode;

    //check for config
    try {
      state.tailwindConfigPath = resolveTailwindConfigPath(config.configPath); //<==
```

**Comment by Arthie** [comment]
@davidtkramer Thank you!

**Comment by davidtkramer** [comment]
@Arthie Sure thing! Thanks for getting this merged! You may already be aware, but just a heads up that I noticed one failing test:

Screen Shot 2020-05-09 at 5 52 06 PM

**Comment by Arthie** [comment]
@davidtkramer Oh didn't see that one! Thanks, I fixed it! v0.5.0 is now on npm 🎉

**Pull request merged**